components/markdown/markdown.tsx

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import messageHtmlToComponent from 'utils/message_html_to_component';
+import EmojiMap from 'utils/emoji_map';
+import * as TextFormatting from 'utils/text_formatting';
+
+type Options = {
```

Comment:
And this can use the `TextFormattingOptions` type from `utils/text_formatting` as well